### ISSUE TITLE
OMP 3.1.2 New releases and featured not visible in front page

### ISSUE THREAD

**Issue by ajnyga** [pkp/pkp-lib#4578]
With a clean install of OMP 3.1.2 I add two new titles, add catalog entries and select `Display featured books on the home page` and  `Display new releases on the home page` from the settings. I then go to my Catalog settings and select Featured and New Release for both titles.

However, no books appear on the front page when using the default theme.

Also the `catalog/newReleases` page is empty.

The books do appear on the public main Catalog page and under their Series page. I can access the abstract page of both titles.

**Comment by ajnyga** [comment]
This is a multi-context issue and connected to the grid that is used to select submissions for new releases and featured content.

In the  `new_releases` table `assoc_id` should be the press id. However, at the moment the code always saves the value `1`.

If I edit the db table by hand and change the assoc_id to match the press id, everything in frontend works.

**Comment by ajnyga** [comment]
Probably due to this: https://github.com/pkp/ui-library/blob/master/src/components/ListPanel/submissions/CatalogSubmissionsListPanel.vue#L237

**Comment by ajnyga** [comment]
There is another reference to "OMP can only have one press" here: https://github.com/pkp/omp/blob/master/classes/install/Upgrade.inc.php#L646

Is this really something that is going to happen with OMP?

**Comment by asmecher** [comment]
No, we have no plans to change OMP to single-press! @bozana, would you mind looking at this?

**Comment by NateWr** [comment]
> No, we have no plans to change OMP to single-press!

@asmecher I'm confused, I thought OMP was _always_ single-press! My assumption here goes way back. 

For example, there's no [site index template file](https://github.com/pkp/omp/tree/master/templates/frontend/pages). You can't access any of the site-wide settings unless a second press exists, and even then they are mostly useless, since most of them would be displayed on a site-wide index page that doesn't exit.

For this reason, the site-wide admin settings forms were _removed_ in OMP with the form refactor. That's also why certain assumptions have been made about the `assoc_id` for new releases and features.

**Comment by ajnyga** [comment]
(I had a discussion concerning the site index with Alec here https://forum.pkp.sfu.ca/t/omp-site-index-redirects-to-login-page/43392)

**Comment by NateWr** [comment]
Ok, we chatted about this and the multi-press stuff needs to come back (https://github.com/pkp/pkp-lib/issues/4762). That also means the `CatalogSubmissionsListPanel.vue` needs to be updated to use the correct context ID.

**Comment by asmecher** [comment]
@NateWr, is this something you could write up in a little more detail to pass off to someone else?

**Comment by asmecher** [comment]
This is a major issue, but is holding up the 3.1.2-2 release unnecessarily -- it may occasion a 3.1.2-3 release when it's ready.

**Comment by NateWr** [comment]
This is the offending line of code:

```
// in OMP, there's only one press context and it's always 1
return 1;
```

https://github.com/pkp/ui-library/blob/master/src/components/ListPanel/submissions/CatalogListPanel.vue#L323

I _think_ all that needs to happen is for the current context ID to be passed into the component as a `data`, and then used there. This would require:

- [ ] In `CatalogSubmissionsListHandler.inc.php`, add `contextId` to the returned array in `getConfig`.
- [ ] In `CatalogSubmissionsListPanel.vue`, add `contextId` to the `data()` method and initialize it with a `0`.
- [ ] In the line above use the data property by replacing `1` with `this.contextId`.

**Comment by asmecher** [comment]
Thanks, @NateWr, I had expected huge tranches of code to need re-adding -- simple enough and confirmed works. I'm having some other issues with catalog management but they are separate and I'll work on them elsewhere.

**Comment by asmecher** [comment]
For OMP 3.1.2-x before 3.1.2-2: https://github.com/pkp/ui-library/commit/d03b95e39bd35825477de27887069b11b7b3bf6b and https://github.com/pkp/omp/commit/1ebc703a2354ebc51f7e8c0192981af259ce15bd need to be applied.

Once that is done, the vue.js code needs to be recompiled using `npm run build` in the OMP installation directory.

**Comment by ajnyga** [comment]
Another context bug in OMP:

https://github.com/pkp/omp/blob/master/classes/workflow/EditorDecisionActionsManager.inc.php#L71

https://github.com/pkp/ojs/blob/master/classes/workflow/EditorDecisionActionsManager.inc.php

In OMP that function does not have a context, but pkp-lib is calling it with a context. This basically breaks the authorDashboard and does not show the review rounds for the authors at all.

Definitely needs to be fixed in the next 3.1.2 release.

**Comment by NateWr** [comment]
Can you file this as a separate issue so that we can keep it separate from this one?

**Comment by ajnyga** [comment]
sure, I was thought it was directly linked to this issue, but I was wrong. I will add a new issue later today.
